lib/catalog/determine_task_relevancy.rb

```diff
@@ -26,36 +26,34 @@ def process
 
     private
 
-    def validate_task
-      return true if order_item
-
-      Rails.logger.info("Incoming task does not belong to this app")
-      false
+    def delegate_task
```

Comment:
I found some differences here, not sure if this is what we want or by accident:
1. no `add_task_update_message`: no need to create `progress messages`?
2. Before `OrderItem` is marked failed only if `status` is `error`, now with an additional `completed state` condition;
3. new `delegate_task` is changed to based on `status`, not `state`: seems like the logic is changed here;

Reply:
@hsong-rh 

1. Yeah, because the underlying `UpdateOrderItem` and `CreateApprovalRequest` are doing that.
2. Yes, I think that if a task errors but is still running, there could be a chance it finishes with a status of "ok". I'm not 100% sure that this can happen, but *if* it can, then it doesn't make sense to mark the order item as failed until the task is actually complete.
3. Right, that's intentional.